homeassistant/components/control4/__init__.py

```diff
@@ -135,30 +98,161 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def get_items_of_category(hass: HomeAssistant, entry: ConfigEntry, category: str):
     """Return a list of all Control4 items with the specified category."""
-    director_all_items = hass.data[DOMAIN][entry.entry_id][CONF_DIRECTOR_ALL_ITEMS]
-    return_list = []
-    for item in director_all_items:
-        if "categories" in item and category in item["categories"]:
-            return_list.append(item)
-    return return_list
+    _LOGGER.debug("Getting items of category: %s", category)
+    director = hass.data[DOMAIN][entry.entry_id][CONF_DIRECTOR]
+    return_list = await director.getAllItemsByCategory(category)
+    return json.loads(return_list)
+
 
+async def refresh_tokens(hass: HomeAssistant, entry: ConfigEntry):
+    """Store updated authentication and director tokens in hass.data, and schedule next token refresh."""
+    config = entry.data
+    verify_ssl_session = aiohttp_client.async_get_clientsession(hass)
 
-class Control4Entity(CoordinatorEntity):
+    account = C4Account(
+        config[CONF_USERNAME], config[CONF_PASSWORD], verify_ssl_session
+    )
+    try:
+        await account.getAccountBearerToken()
+    except client_exceptions.ClientError as exception:
+        _LOGGER.error("Error connecting to Control4 account API: %s", exception)
+        raise ConfigEntryNotReady from exception
+    except BadCredentials as exception:
+        _LOGGER.error(
+            "Error authenticating with Control4 account API, incorrect username or password: %s",
+            exception,
+        )
```

Comment:
```suggestion
```

Similar feedback to https://github.com/home-assistant/core/pull/60465/files#r780069119